spotbugs-tests/src/test/java/edu/umd/cs/findbugs/detect/BadVisibilityOnSharedPrimitiveVariablesTest.java

```diff
@@ -0,0 +1,100 @@
+package edu.umd.cs.findbugs.detect;
+
+import edu.umd.cs.findbugs.AbstractIntegrationTest;
+import edu.umd.cs.findbugs.SystemProperties;
+import edu.umd.cs.findbugs.test.matcher.BugInstanceMatcher;
+import edu.umd.cs.findbugs.test.matcher.BugInstanceMatcherBuilder;
+import org.junit.Test;
+
+import static edu.umd.cs.findbugs.test.CountMatcher.containsExactly;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.hasItem;
+
+public class BadVisibilityOnSharedPrimitiveVariablesTest extends AbstractIntegrationTest {
+
+    @Test
+    public void failurePath_fieldWithBadVisibility_whenOtherMethodHasSynchronizedBlock() {
+        SystemProperties.setProperty("ful.debug", "true");
```

Comment:
Perhaps this should be in a pre-test setup method?

Reply:
It is not even really needed, only if someone wants to check the logs. I moved it to a pre-test method though, maybe sometime it is going to be useful for someone.